src/cdk/overlay/dispatchers/overlay-keyboard-dispatcher.spec.ts

```diff
@@ -178,6 +183,21 @@ describe('OverlayKeyboardDispatcher', () => {
     expect(overlayTwoSpy).not.toHaveBeenCalled();
     expect(overlayOneSpy).toHaveBeenCalled();
   });
+
+  it('should not run change detection if there are no `keydownEvents` observers', () => {
+    spyOn(appRef, 'tick');
+    const overlayRef = overlay.create();
+    keyboardDispatcher.add(overlayRef);
+
+    expect(appRef.tick).toHaveBeenCalledTimes(0);
+    dispatchKeyboardEvent(document.body, 'keydown', ESCAPE);
+    expect(appRef.tick).toHaveBeenCalledTimes(0);
+
+    overlayRef.keydownEvents().subscribe();
+    dispatchKeyboardEvent(document.body, 'keydown', ESCAPE);
+
+    expect(appRef.tick).toHaveBeenCalledTimes(1);
```

Comment:
Are you sure that this test is checking the correct behavior? I tried copying it into master and it fails on this line (199). Shouldn't it have failed on line 194 since previously it would've triggered a `tick`?

Reply:
Well, I see what you mean. This fails on master because tests are being run within the `<root>` zone (which differs from the app behavior):
```ts
    spyOn(appRef, 'tick');
    const overlayRef = overlay.create();
    keyboardDispatcher.add(overlayRef);
    console.log(Zone!.current); // <root>
```

Basically, `add` is being called outside of the Angular zone within unit tests, but it's being called within Angular zone in actual apps.

Reply:
Alright, that makes sense.